internal/controller/deployment.go

```diff
@@ -121,6 +118,15 @@ func ConstructEnvForApp(ctx context.Context, app *spinv1alpha1.SpinApp) []corev1
 		envs[idx] = env
 	}
 
+	// Adding the OpenTelemetry params
+	for _, entry := range app.Spec.Otel {
+		envs = append(envs, corev1.EnvVar{Name: entry.Name, Value: entry.Value, ValueFrom: entry.ValueFrom})
+	}
+
+	if len(envs) == 0 {
```

Comment:
Why is this here?

Reply:
The original buildEnvs had a conditional at the beginning which returned `nil` if there were no spin variables defined. I assumed that returning an empty `[]corev1.EnvVar` would be a problem, so I returned `nil` if neither spin nor otel variables were defined. Sounds like it might not be necessary?

Reply:
Not sure if it is necessary, but your logic is reasonable to me. Let's leave it like this.